src/Avalonia.Base/Controls/ResourceDictionary.cs

```diff
@@ -147,6 +147,12 @@ public void AddDeferred(object key, Func<IServiceProvider?, object?> factory)
         public void AddDeferred(object key, IDeferredContent deferredContent)
             => Add(key, deferredContent);
 
+        public void AddNotShared(object key, Func<IServiceProvider?, object?> factory)
+            => Add(key, new NotSharedDeferredItem(factory));
```

Comment:
We have AddDeferred with a factory delegate only because it was first added. For AddNotShared there is no need for that.